stylelint-polaris/plugins/coverage/index.js

```diff
@@ -2,79 +2,114 @@ const stylelint = require('stylelint');
 
 const {isPlainObject} = require('../../utils');
 
-const ruleName = 'polaris/coverage';
+const coverageRuleName = 'polaris/coverage';
+
+/**
+ * @typedef {import('stylelint').ConfigRules} StylelintRuleConfig
+ * @property {string} message - Message appended to the warning in place of the default category message
+ */
+
+/**
+ * @typedef {object} CategorySettings
+ * @property {import('stylelint').RuleMessage} [message] - Message appended to the warning if no custom message is set on a rule's secondary options
+ * @property {import('stylelint').RuleMeta} [meta] - Category documentation URL hyperlinked to the reported rule in the VS Code diagnostic
+ */
 
 /**
  * @typedef {{
- *   [category: string]: import('stylelint').ConfigRules
- * }} PrimaryOptions
+ *   [category: string]: StylelintRuleConfig | [
+ *     StylelintRuleConfig, CategorySettings
+ *   ]
+ * }} CategorizedRules
  */
 
 // Setting `line` to an invalid line number forces the warning to be reported
 // and the `report({node})` option is used to display the location information:
 // https://github.com/stylelint/stylelint/blob/57cbcd4eb0ee809006a1e3d2ccfe73af48744ad5/lib/utils/report.js#L49-L52
 const forceReport = {line: -1};
+const defaultMeta = {
+  url: 'https://github.com/Shopify/polaris/tree/main/stylelint-polaris/plugins/coverage/README.md#rules',
+};
 
 module.exports = stylelint.createPlugin(
-  ruleName,
-  /** @param {PrimaryOptions} primaryOptions */
-  (primaryOptions, secondaryOptions, context) => {
-    const isPrimaryOptionsValid = validatePrimaryOptions(primaryOptions);
-
-    const rules = [];
-
-    for (const [categoryName, categoryConfigRules] of Object.entries(
-      primaryOptions,
-    )) {
-      for (const [categoryRuleName, categoryRuleSettings] of Object.entries(
-        categoryConfigRules,
+  coverageRuleName,
+  /**
+   * @param {CategorizedRules} categorizedRules - Configured Stylelint rules grouped by Polaris coverage category
+   * @param {import('stylelint').RuleContext} context
+   */
+  (categorizedRules, _, context) => {
+    const isPrimaryOptionsValid = validatePrimaryOptions(categorizedRules);
+    const coverageRules = [];
+
+    for (const [category, categoryConfig] of Object.entries(categorizedRules)) {
+      const [stylelintRules, categorySettings] =
+        normalizeConfig(categoryConfig);
+
+      for (const [stylelintRuleName, stylelintRuleConfig] of Object.entries(
```

Comment:
Same as above, I think this needs to be disambiguated as we are iterating over custom and built-in Stylelint rules (not just Stylelint rules). Also, the updated names are now diverging from Stylelint's [type definitions](https://github.com/stylelint/stylelint/blob/11acf312191e5097692f7428f13977bd1fd12cd4/types/stylelint/index.d.ts#L17) (which the plugin is heavily tide too) and is becoming an additional layer of mental overhead on top of the existing Stylelint terminology.